GVFS/GVFS.UnitTests/Service/MacServiceTests.cs

```diff
@@ -0,0 +1,137 @@
+﻿using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using GVFS.Platform.Mac;
+using GVFS.Service;
+using GVFS.UnitTests.Mock.Common;
+using GVFS.UnitTests.Mock.FileSystem;
+using Moq;
+using NUnit.Framework;
+using System.IO;
+
+namespace GVFS.UnitTests.Service
+{
+    [TestFixture]
+    [NonParallelizable]
+    public class MacServiceTests
+    {
+        private const string GVFSServiceName = "GVFS.Service";
+        private const string ExpectedActiveUserId = "502";
+        private static readonly string ExpectedActiveRepoPath = Path.Combine("mock:", "code", "repo2");
+        private static readonly string ServiceDataLocation = Path.Combine("mock:", "registryDataFolder");
+
+        private MockFileSystem fileSystem;
+        private MockTracer tracer;
+        private Mock<MacPlatform> gvfsPlatformMock;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.tracer = new MockTracer();
+            this.fileSystem = new MockFileSystem(new MockDirectory(ServiceDataLocation, null, null));
+            this.gvfsPlatformMock = new Mock<MacPlatform>();
+            this.gvfsPlatformMock.Setup(p => p.GetCurrentUser()).Returns(ExpectedActiveUserId);
+            this.gvfsPlatformMock.Setup(p => p.GetUserIdFromLoginSessionId(It.IsAny<int>())).Returns<int>(x => x.ToString());
+            this.gvfsPlatformMock.SetupGet(p => p.FileSystem).Returns(new MockPlatformFileSystem());

```

Comment:
Why are we using a `MockPlatformFileSystem` here but then the `this.fileSystem` is a `MockFileSystem` what are the differences and why are we using 2 different ones?

Reply:
Differences - 

|`MockPlatformFileSystem`|`MockFileSystem`|
|---------------------------|------------------|
|Used to Mock `IPlatformFileSystem`|Mocks `PhysicalFileSystem`|
|Tries to encapsulate platform specific methods|Platform neutral (internally uses `GVFSPlatform` in some methods though)|
|Used by `RepoRegistry` to get Normalized file path on the target OS platform|Used by `RepoRegistry` to read & write files|

The two classes are Not interchangeable.  `RepoRegistry` needs `PhysicalFileSystem` and `GVFSPlatform`. `GVFSPlatform` needs `IPlatformFileSystem`.